frame/support/src/dispatch.rs

```diff
@@ -1787,6 +1812,31 @@ macro_rules! decl_module {
 			}
 		}
 
+		// Bring `GetPalletVersion` into scope to make it easily usable.
+		pub use $crate::traits::GetPalletVersion as _;
+		// Implement `GetPalletVersion` for `Module`
+		impl<$trait_instance: $trait_name $(<I>, $instance: $instantiable)?> $crate::traits::GetPalletVersion
+			for $mod_type<$trait_instance $(, $instance)?> where $( $other_where_bounds )*
+		{
+			fn current_version() -> $crate::traits::PalletVersion {
+				$crate::crate_to_pallet_version!()
+			}
+
+			fn storage_version() -> $crate::traits::PalletVersion {
+				let key = $crate::traits::PalletVersion::storage_key::<
+						<$trait_instance as $system::Trait>::PalletInfo, Self
+					>().expect("Every active pallet has a name in the runtime; qed");
+
+				let value: Option<$crate::traits::PalletVersion> =
+					$crate::storage::unhashed::get(&key);
+
+				match value {
+					Some(version) => version,
+					None => $crate::crate_to_pallet_version!(),
```

Comment:
I don't think that this default will work well. I want to be able to determine whether a version has been set.
I would either use `0.0.0` as the default or return an option for the storage version and return `None` here. The latter is cleaner and preferable IMO.